components/src/dynamo/common/utils/video_utils.py

```
@@ -242,3 +245,297 @@ def encode_to_video_bytes(
+        or os.environ.get("DYNAMO_VAAPI_DEVICE")
+        or "/dev/dri/renderD128"
+    )
+
+
+def _running_on_xpu() -> bool:
+    """Return True when torch reports an available XPU backend."""
+    try:
+        import torch
+
+        return bool(hasattr(torch, "xpu") and torch.xpu.is_available())
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)